### PR TITLE
Remove rotation and basisstate

### PR DESCRIPTION
**Pull request by antalszava**
Remove ``gates.py`` such that no PL-Qiskit ``Rot`` and ``Basisstate`` are used anymore. These operations get decomposed based on the implementation in PennyLane core.

Added an integration test for these two operations, such that they are decomposed. In the test, ``analytic=False`` for the device, such that hw simulators are not skipped.

This PR follows up on issue #52.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,85 +1,102 @@
...
+
+
+@pytest.fixture(params=hw_backends)
+def hardware_backend(request):
+    return request.param
+
+
+@pytest.fixture(params=state_backends)
+def statevector_backend(request):
+    return request.param
```
Comment (antalszava) [inline tests/conftest.py]: Added these for convenience in the tests

File: tests/conftest.py [hunk 2]
```diff
@@ -1,85 +1,102 @@
...
+
+    return _device
+
+
+@pytest.fixture(params=[AerDevice, BasicAerDevice])
+def state_vector_device(request, statevector_backend, shots, analytic):
+    def _device(n):
+        return request.param(wires=n, backend=statevector_backend, shots=shots, analytic=analytic)
+
+    return _device
```
Comment (antalszava) [inline tests/conftest.py]: This one as well

File: tests/test_converter.py [hunk 3]
```diff
@@ -327,6 +333,16 @@ def test_one_qubit_parametrized_operations_supported_by_pennylane(self, recorder
...
+        assert recorder.queue[4].name == 'U2'
+        assert len(recorder.queue[4].params) == 2
+        assert recorder.queue[4].params == [0.3, 0.4]
+        assert recorder.queue[4].wires == [0]
+
+        assert recorder.queue[5].name == 'U3'
+        assert len(recorder.queue[5].params) == 3
+        assert recorder.queue[5].params == [0.3, 0.4, 0.2]
+        assert recorder.queue[5].wires == [0]
+
```
Comment (antalszava) [inline tests/test_converter.py]: These were previously in the ``test_operations_transformed_into_qubit_unitary``, now that PL core has them, they could be reorganized

File: tests/test_converter.py [hunk 4]
```diff
@@ -495,19 +507,22 @@ def test_operations_transformed_into_qubit_unitary(self, recorder):
...
-        assert recorder.queue[3].name == 'U3'
-        assert len(recorder.queue[3].params) == 3
-        assert recorder.queue[3].params == [0.3, 0.4, 0.2]
-        assert recorder.queue[3].wires == [0]
+        qc = QuantumCircuit(3, 1)
 
-        assert recorder.queue[4].name == 'Toffoli'
-        assert len(recorder.queue[4].params) == 0
-        assert recorder.queue[4].wires == [0, 1, 2]
+        qc.ch([0], [1])
```
Comment (antalszava) [inline tests/test_converter.py]: Added a test case for an operation from Qiskit that has the ``to_matrix()`` method defined on them and is not in PL core

File: tests/test_inverses.py [hunk 5]
```diff
@@ -136,6 +130,27 @@ def circuit():
...
 
+    @pytest.mark.parametrize("name,par,expected_output", [
+        ("Rot", [math.pi/2, 0, 0], 1),
+        ("Rot", [0, math.pi/2, 0], 0),
+        ("Rot", [0, 0, math.pi/2], 1),
+        ("Rot", [math.pi/2, -math.pi/4, -math.pi/4], 1/math.sqrt(2)),
+        ("Rot", [-math.pi/4, math.pi/2, math.pi/4], 0),
+        ("Rot", [-math.pi/4, math.pi/4, math.pi/2], 1/math.sqrt(2)),
+    ])
+    def test_unsupported_gate_inverses(self, name, par, expected_output):
```
Comment (antalszava) [inline tests/test_inverses.py]: Reorganized this into a separate test case, as ``assert dev.supports_operation("Rot")`` evaluates to False

File: tests/test_converter.py (hunk 4 above)
Comment (josh146) [reply tests/test_converter.py]: nice!

**Pull request merged**